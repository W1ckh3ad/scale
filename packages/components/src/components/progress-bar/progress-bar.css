:host {
  --color-error: var(--scl-color-text-error);
  --color-disabled: var(--scl-color-grey-50);
  --color-primary: var(--scl-color-primary);
  --background-standard: var(--scl-color-background-standard);
  --font-size-inner-status: var(--scl-font-size-12);
  --font-size-status: var(--scl-font-size-12);
  --font-weight-status: var(--scl-font-weight-extrabold);
  --color-status: var(--scl-color-grey-60);
  --color-status-description: var(--scl-color-grey-60);
  --font-size-status-description: var(--scl-font-size-12);
  --font-weight-status-description: var(--scl-font-weight-bold);
  --spacing-label: var(--scl-spacing-8) 0;
  --border: 1px solid var(--scl-color-grey-60);
}

.progress-bar {
  width: 100%;
  max-width: 30rem;
}

.progress-bar--has-error .progress-bar__outer {
  border: 1px solid var(--color-error);
}

.progress-bar--has-error .progress-bar__status-description {
  color: var(--color-error);
}

.progress-bar--has-error scale-icon {
  --icon-color: var(--color-error);
}

.progress-bar--disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.progress-bar--disabled .progress-bar__label {
  color: var(--color-disabled);
}

.progress-bar__label {
  display: block;
  padding: var(--spacing-label);
}

.progress-bar-wrapper {
  width: 100%;
  display: flex;
  box-sizing: border-box;
  align-items: center;
}

.progress-bar__outer {
  width: 100%;
  height: var(--scl-spacing-8);
  margin: var(--scl-spacing-8);
  margin-left: 0;
  overflow: hidden;
  position: relative;
  border-radius: 100px;
  border: var(--border);
}

.progress-bar__inner {
  top: 0;
  left: 0;
  height: 100%;
  display: flex;
  position: absolute;
  align-items: center;
  white-space: nowrap;
  justify-content: flex-end;
  animation-fill-mode: both;
  background: var(--color-primary);
}

.progress-bar__inner-status {
  color: var(--background-standard);
  padding: 0 12px;
  font-size: var(--font-size-inner-status);
}

.progress-bar__status {
  display: inline-block;
  font-size: var(--font-size-status);
  font-weight: var(--font-weight-status);
  color: var(--color-status);
}

.progress-bar__icon {
  height: 16px;
}

.progress-bar__status-description {
  color: var(--color-status-description);
  font-size: var(--font-size-status-description);
  font-weight: var(--font-weight-status-description);
}

.progress-bar__aria-live {
  clip: rect(0 0 0 0);
  width: 1px;
  border: 0;
  height: 1px;
  margin: -1px;
  padding: 0;
  overflow: hidden;
  position: absolute;
}
