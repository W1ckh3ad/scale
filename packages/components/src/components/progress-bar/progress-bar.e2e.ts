import { newE2EPage } from '@stencil/core/testing';

describe('scale-progress-bar', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<scale-progress-bar/>');
    const element = await page.find('scale-progress-bar');
    expect(element).toHaveClass('hydrated');
  });
});
