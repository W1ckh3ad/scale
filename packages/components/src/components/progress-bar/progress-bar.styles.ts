import { JssStyle } from 'jss';

export const styles: JssStyle = {
  'progress-bar': {
    width: '100%',
  },
  'progress-bar--has-error': {
    '& $progress-bar__outer': {
      border: ({ colors }) => `1px solid ${colors.error.default}`,
    },
    '& $progress-bar__status-description': {
      color: ({ colors }) => `${colors.error.default}`,
    },
    '& scale-icon': {
      '--icon-color': ({ colors }) => `${colors.error.default}`,
    },
  },
  'progress-bar--disabled': {
    cursor: 'not-allowed',
    opacity: 0.5,
  },
  'progress-bar-wrapper': {
    display: 'flex',
    width: '100%',
    alignItems: 'center',
    boxSizing: 'border-box',
    '& scale-icon': {
      marginTop: 12,
    },
  },
  'progress-bar__outer': {
    margin: '12px 12px 12px 0',
    position: 'relative',
    height: '8px',
    borderRadius: '100px',
    background: '#ebeef5',
    overflow: 'hidden',
    width: '100%',
  },
  'progress-bar__inner': {
    position: 'absolute',
    left: 0,
    top: 0,
    height: '100%',
    borderRadius: '100px',
    whiteSpace: 'nowrap',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    animationFillMode: 'both',
    background: ({ colors }) => `${colors.primary.default}`,
  },
  'progress-bar__inner-status': {
    color: '#fff',
    fontSize: '12px',
    padding: '0 12px',
  },
  'progress-bar__status': {
    color: '#606266',
    fontSize: '12px',
    display: 'inline-block',
  },
  'progress-bar__status-description': {
    color: '#606266',
    fontSize: '12px',
  },
};
