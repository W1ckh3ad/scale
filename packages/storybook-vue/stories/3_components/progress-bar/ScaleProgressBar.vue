<template>
    <scale-progress-bar
      :icon="icon"
      :custom-class="customClass"
      :custom-color="customColor"
      :disabled="disabled"
      :has-error="hasError"
      :status-inside="statusInside"
      :status-description="statusDescription"
      :show-status="showStatus"
      :stroke-width="strokeWidth"
      :percentage="percentage"
      :styles="styles"
    >
    </scale-progress-bar>
</template>

<script>
  export default {
    props: {
      icon: String,
      customClass: String,
      customColor: String,
      disabled: { type: Boolean, default: false },
      statusInside: Boolean,
      hasError: Boolean,
      statusDescription: String,
      styles: String,
      showStatus: { type: Boolean, default: true },
      strokeWidth: Number,
      percentage: Number,
    }
  }
</script>




