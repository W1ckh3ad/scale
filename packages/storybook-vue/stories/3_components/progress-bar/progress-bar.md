# Progress Bar

The progress bar informs users about the status of an ongoing process, such as loading an app, submitting a form or saving updates.

## Basic Types

### Determinate

Determinate indicators display how long a process will take. They should be used when it is possible to define the process completion rate.

![Image Name](assets/3_components/progress-bar/image-20200810094334611.png)

### Indeterminate

Indeterminate indicators illustrate an unspecified amount of wait time. They should be used when it is either not possible or not necessary to indicate how long a process will take.

![Image Name](assets/3_components/progress-bar/image-20200810094322804.png)

## Elements

### Description (1) (Optional)

### Status (2) (Optional)

![Image Name](assets/3_components/progress-bar/image-20200810094345964.png)

## Colors

Use the default color or assign a custom color. When customizing, use a color-coding scheme that is clear and supports the user in what they want to achieve.

### Progress

#### Default

![Image Name](assets/3_components/progress-bar/image-20200810094334611.png)

#### Custom

![Image Name](assets/3_components/progress-bar/image-20200810094258891.png)

### Complete

#### Default

![Image Name](assets/3_components/progress-bar/image-20200811100931435.png)

#### Green

![Image Name](assets/3_components/progress-bar/image-20200811100938570.png)
#### Error

![Image Name](assets/3_components/progress-bar/image-20200811100945283.png)
