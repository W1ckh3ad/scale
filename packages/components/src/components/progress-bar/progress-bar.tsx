import { Component, Prop, h, Host } from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';
import { styles } from './progress-bar.styles';
import { CssInJs } from '../../utils/css-in-js';
import { StyleSheet } from 'jss';
import Base from '../../utils/base-interface';

let i = 0;

@Component({
  tag: 'scale-progress-bar',
  shadow: true,
})
export class ProgressBar implements Base {
  /** (optional) Progress bar class */
  @Prop() customClass?: string = '';
  /** (optional) Progress bar busy switch */
  @Prop() busy?: boolean = false;
  /** (required) Progress bar percentage */
  @Prop() percentage: number = 0;
  /** (optional) Progress bar customColor */
  @Prop() customColor?: string;
  /** (optional) Progress bar stroke width */
  @Prop() strokeWidth?: number = 6;
  /** (optional) Progress bar percentage text */
  @Prop() showStatus?: boolean;
  /** (optional) Progress bar icon indicator */
  @Prop() icon?: string;
  /** (optional) Progress bar status description text */
  @Prop() statusDescription?: string;
  /** (optional) Progress text display inside bar */
  @Prop() statusInside?: boolean;
  /** (optional) Progress bar error */
  @Prop() hasError?: boolean;
  /** (optional) Progress bar disabled */
  @Prop() disabled?: boolean;
  /** (optional) Progress bar id */
  @Prop() progressBarId?: string;
  /** (optional) Progress bar label */
  @Prop() label?: string;
  /** (optional) Injected jss styles */
  @Prop() styles?: any;
  /** decorator Jss stylesheet */
  @CssInJs('ProgressBar', styles) stylesheet: StyleSheet;

  componentWillLoad() {
    if (this.progressBarId == null) {
      this.progressBarId = 'progress-bar-' + i++;
    }
  }
  componentWillUpdate() {}
  componentDidUnload() {}

  transitions = (width: number) => `
    @keyframes showProgress {
      from {
        width: 0;
      }
      to {
        width: ${width}%;
      }
    }
  `;

  progressStyle = () => {
    return {
      width: `${this.percentage}%`,
      border: '1px solid transparent',
      background: this.customColor,
      animation: 'showProgress 3s ease-in-out',
    };
  };

  render() {
    const { classes } = this.stylesheet;

    return (
      <Host>
        <style>{this.stylesheet.toString()}</style>
        <style>{this.transitions(this.percentage)}</style>
        <div class={this.getCssClassMap()}>
          {!!this.label && (
            <label
              class={classes['progress-bar__label']}
              htmlFor={this.progressBarId}
            >
              {this.label}
            </label>
          )}
          <div class={classes['progress-bar-wrapper']}>
            <div
              class={classes['progress-bar__outer']}
              style={{ height: `${this.strokeWidth}px` }}
              role="progressbar"
              aria-valuemin={0}
              aria-valuemax={100}
              aria-valuenow={this.percentage}
              aria-busy={this.busy}
              aria-valuetext={`${this.percentage}%`}
              aria-label={this.label}
              id={this.progressBarId}
            >
              <div
                class={classes['progress-bar__inner']}
                style={this.progressStyle()}
              >
                {!!this.statusInside && (
                  <div
                    class={classes['progress-bar__inner-status']}
                    aria-hidden="true"
                  >{`${this.percentage}%`}</div>
                )}
              </div>
            </div>

            {!!this.showStatus && (
              <div
                class={classes['progress-bar__status']}
                aria-hidden="true"
              >{`${this.percentage}%`}</div>
            )}
            {!!this.icon && (
              <scale-icon
                class={classes['progress-bar__status']}
                path={this.icon}
                size={16}
              ></scale-icon>
            )}
          </div>
        </div>
        {!!this.statusDescription && (
          <div class={classes['progress-bar__status-description']} role="alert">
            {this.statusDescription}
          </div>
        )}
        {
          <span aria-live="polite" class={classes['progress-bar__aria-live']}>
            {this.percentage !== Math.round(this.percentage / 10) * 10
              ? `${Math.round(this.percentage / 10) * 10}%`
              : null}
          </span>
        }
      </Host>
    );
  }

  getCssClassMap(): CssClassMap {
    const { classes } = this.stylesheet;

    return classNames(
      classes['progress-bar'],
      this.customClass && this.customClass,
      this.hasError && classes[`progress-bar--has-error`],
      this.disabled && classes[`progress-bar--disabled`]
    );
  }
}
