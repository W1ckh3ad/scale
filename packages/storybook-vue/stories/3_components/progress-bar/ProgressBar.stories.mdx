import { Meta, Story, Props, Preview, Description } from '@storybook/addon-docs/blocks';
import ScaleProgressBar from './ScaleProgressBar.vue';
import description from './progress-bar.md';
import { action } from "@storybook/addon-actions";
export const ICON = 'M8,0.666666667 C12.05,0.666666667 15.3333333,3.95 15.3333333,8 C15.3333333,12.05 12.05,15.3333333 8,15.3333333 C3.95,15.3333333 0.666666667,12.05 0.666666667,8 C0.666666667,3.95 3.95,0.666666667 8,0.666666667 Z M8,1.66666667 C4.50766667,1.66666667 1.66666667,4.50766667 1.66666667,8 C1.66666667,11.4923333 4.50766667,14.3333333 8,14.3333333 C11.4923333,14.3333333 14.3333333,11.4923333 14.3333333,8 C14.3333333,4.50766667 11.4923333,1.66666667 8,1.66666667 Z M10.7107867,5.28921328 C10.8882978,5.46672432 10.9044351,5.74450011 10.7591988,5.9402411 L10.7107867,5.99632006 L8.70666667,7.99966667 L10.7107867,10.0036799 C10.9060489,10.1989421 10.9060489,10.5155246 10.7107867,10.7107867 C10.5332757,10.8882978 10.2554999,10.9044351 10.0597589,10.7591988 L10.0036799,10.7107867 L7.99966667,8.70666667 L5.99645339,10.7106534 C5.80119124,10.9059155 5.48460876,10.9059155 5.28934661,10.7106534 C5.11183557,10.5331423 5.0956982,10.2553666 5.24093451,10.0596256 L5.28934661,10.0035466 L7.29266667,7.99966667 L5.28934661,5.99645339 C5.09408446,5.80119124 5.09408446,5.48460876 5.28934661,5.28934661 C5.46685765,5.11183557 5.74463344,5.0956982 5.94037444,5.24093451 L5.99645339,5.28934661 L7.99966667,7.29266667 L10.0036799,5.28921328 C10.1989421,5.09395113 10.5155246,5.09395113 10.7107867,5.28921328 Z';

<Meta
  title="Components/ProgressBar"
  component={ScaleProgressBar}
  argTypes={{}}
/>

export const Template = (args, {argTypes}) => ({
  components:{ ScaleProgressBar },
  props: {
    ...ScaleProgressBar.props
  },
  template: `
  <scale-progress-bar
    :icon="icon"
    :custom-class="customClass"
    :custom-color="customColor"
    :disabled="disabled"
    :has-error="hasError"
    :status-inside="statusInside"
    :status-description="statusDescription"
    :show-status="showStatus"
    :stroke-width="strokeWidth"
    :percentage="percentage"
    :styles="styles"
  >
  </scale-progress-bar>
  `
})

<Description>{description}</Description>

<Preview withSource="none">
  <Story
    name="Standard"
    args={{
      percentage: 20,
    }}
  >
    {Template.bind({})}
  </Story>
</Preview>

<Props story='Standard' />


## ProgressBar Track Small

<Preview withSource="none">
  <Story
    name="ProgressBar Track Small"
    args={{
      percentage: 20,
      strokeWidth: 2,
    }}
  >
    {Template.bind({})}
  </Story>
</Preview>

```html
<scale-progress-bar :percentage="20" :track-height="2"></scale-progress-bar>
```


## ProgressBar With Custom Color

<Preview withSource="none">
  <Story
    name="ProgressBar With Custom Color"
    args={{
      percentage: 20,
      customColor: "#009DE0"
    }}
  >
    {Template.bind({})}
  </Story>
</Preview>

```html
<scale-progress-bar :percentage="20" :custom-color="#009DE0"></scale-progress-bar>
```


## Disabled ProgressBar

<Preview withSource="none">
  <Story
    name="Disabled ProgressBar"
    args={{
      percentage: 40,
      disabled: true
    }}
  >
    {Template.bind({})}
  </Story>
</Preview>

```html
<scale-progress-bar :percentage="40" :disabled="true"></scale-progress-bar>
```


## Progress Percentage Inside Bar

<Preview withSource="none">
  <Story
    name="Progress Percentage Inside Bar"
    args={{
      percentage: 40,
      strokeWidth: 10,
      statusInside: true,
      showStatus: false,
    }}
  >
    {Template.bind({})}
  </Story>
</Preview>

```html
<scale-progress-bar :percentage="40" :stroke-width="10" :status-inside="true" :show-status="false"></scale-progress-bar>
```


## ProgressBar With Description

<Preview withSource="none">
  <Story
    name="ProgressBar With Description"
    args={{
      percentage: 40,
      statusDescription: 'ProgressBar status description text'
    }}
  >
    {Template.bind({})}
  </Story>
</Preview>

```html
<scale-progress-bar :percentage="40" :stroke-width="40" :status-description="ProgressBar status description text"></scale-progress-bar>
```


## ProgressBar With Error and Icon

<Preview withSource="none">
  <Story
    name="ProgressBar With Error and Icon"
    args={{
      percentage: 40,
      hasError: true,
      showStatus: false,
      icon: ICON,
    }}
  >
    {Template.bind({})}
  </Story>
</Preview>

```html
<scale-progress-bar :percentage="40" :has-error="true" :icon="SVG-PATH" :show-status="false"></scale-progress-bar>
```
