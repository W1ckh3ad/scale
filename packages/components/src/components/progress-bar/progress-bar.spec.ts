import { newSpecPage } from '@stencil/core/testing';
import { ProgressBar } from './progress-bar';
import { styles } from './progress-bar.styles';
import jss from 'jss';

describe('ProgressBar', () => {
  let element;
  let stylesheet;
  beforeEach(async () => {
    element = new ProgressBar();
    stylesheet = element.stylesheet = jss.createStyleSheet(styles as any);
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [ProgressBar],
      html: `<scale-progress-bar>Label</scale-progress-bar>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should have css property width 24px when stroke width is set to 24', async () => {
    const page = await newSpecPage({
      components: [ProgressBar],
      html: `<scale-progress-bar stroke-width=24>Label</scale-progress-bar>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should contain progress-bar-text css class when show text is set true', async () => {
    const page = await newSpecPage({
      components: [ProgressBar],
      html: `<scale-progress-bar show-text=true>Label</scale-progress-bar>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should contain progress-bar-inner-text css class when text inside is set true', async () => {
    const page = await newSpecPage({
      components: [ProgressBar],
      html: `<scale-progress-bar text-inside=true>Label</scale-progress-bar>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle css classes', () => {
    element.customClass = 'custom';
    expect(element.getCssClassMap()).toContain('custom');

    element.variant = 'primary';
    stylesheet.addRule('progress-bar--variant-primary', {});
    expect(element.getCssClassMap()).toContain(
      stylesheet.classes['progress-bar--variant-primary']
    );
  });
});
