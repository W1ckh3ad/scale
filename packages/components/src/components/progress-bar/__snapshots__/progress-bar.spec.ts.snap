// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should contain progress-bar-inner-text css class when text inside is set true 1`] = `
<scale-progress-bar text-inside="true">
  <mock:shadow-root>
    <style>
      @keyframes showProgress {
      from {
        width: 0;
      }
      to {
        width: 0%;
      }
    }
    </style>
    <div class="progress-bar" part="progress-bar">
      <div class="progress-bar-wrapper" part="wrapper">
        <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="0" aria-valuetext="0%" class="progress-bar__outer" id="progress-bar-7" part="outer" role="progressbar" style="height: 6px;">
          <div class="progress-bar__inner" part="inner" style="width: 0%; border: 1px solid transparent; animation: showProgress 3s ease-in-out;"></div>
        </div>
        <slot name="icon"></slot>
      </div>
    </div>
    <span aria-live="polite" class="progress-bar__aria-live"></span>
  </mock:shadow-root>
  Label
</scale-progress-bar>
`;

exports[`ProgressBar should contain progress-bar-text css class when show text is set true 1`] = `
<scale-progress-bar show-status="true">
  <mock:shadow-root>
    <style>
      @keyframes showProgress {
      from {
        width: 0;
      }
      to {
        width: 0%;
      }
    }
    </style>
    <div class="progress-bar" part="progress-bar">
      <div class="progress-bar-wrapper" part="wrapper">
        <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="0" aria-valuetext="0%" class="progress-bar__outer" id="progress-bar-6" part="outer" role="progressbar" style="height: 6px;">
          <div class="progress-bar__inner" part="inner" style="width: 0%; border: 1px solid transparent; animation: showProgress 3s ease-in-out;"></div>
        </div>
        <div aria-hidden="true" class="progress-bar__status" part="status">
          0%
        </div>
        <slot name="icon"></slot>
      </div>
    </div>
    <span aria-live="polite" class="progress-bar__aria-live"></span>
  </mock:shadow-root>
  Label
</scale-progress-bar>
`;

exports[`ProgressBar should have css property width 24px when stroke width is set to 24 1`] = `
<scale-progress-bar stroke-width="24">
  <mock:shadow-root>
    <style>
      @keyframes showProgress {
      from {
        width: 0;
      }
      to {
        width: 0%;
      }
    }
    </style>
    <div class="progress-bar" part="progress-bar">
      <div class="progress-bar-wrapper" part="wrapper">
        <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="0" aria-valuetext="0%" class="progress-bar__outer" id="progress-bar-5" part="outer" role="progressbar" style="height: 24px;">
          <div class="progress-bar__inner" part="inner" style="width: 0%; border: 1px solid transparent; animation: showProgress 3s ease-in-out;"></div>
        </div>
        <slot name="icon"></slot>
      </div>
    </div>
    <span aria-live="polite" class="progress-bar__aria-live"></span>
  </mock:shadow-root>
  Label
</scale-progress-bar>
`;

exports[`ProgressBar should match snapshot 1`] = `
<scale-progress-bar>
  <mock:shadow-root>
    <style>
      @keyframes showProgress {
      from {
        width: 0;
      }
      to {
        width: 0%;
      }
    }
    </style>
    <div class="progress-bar" part="progress-bar">
      <div class="progress-bar-wrapper" part="wrapper">
        <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="0" aria-valuetext="0%" class="progress-bar__outer" id="progress-bar-0" part="outer" role="progressbar" style="height: 6px;">
          <div class="progress-bar__inner" part="inner" style="width: 0%; border: 1px solid transparent; animation: showProgress 3s ease-in-out;"></div>
        </div>
        <slot name="icon"></slot>
      </div>
    </div>
    <span aria-live="polite" class="progress-bar__aria-live"></span>
  </mock:shadow-root>
  Label
</scale-progress-bar>
`;

exports[`ProgressBar should match snapshot with label set 1`] = `
<scale-progress-bar label="testLabel">
  <mock:shadow-root>
    <style>
      @keyframes showProgress {
      from {
        width: 0;
      }
      to {
        width: 0%;
      }
    }
    </style>
    <div class="progress-bar" part="progress-bar">
      <label class="progress-bar__label" htmlfor="progress-bar-1" part="label">
        testLabel
      </label>
      <div class="progress-bar-wrapper" part="wrapper">
        <div aria-label="testLabel" aria-valuemax="100" aria-valuemin="0" aria-valuenow="0" aria-valuetext="0%" class="progress-bar__outer" id="progress-bar-1" part="outer" role="progressbar" style="height: 6px;">
          <div class="progress-bar__inner" part="inner" style="width: 0%; border: 1px solid transparent; animation: showProgress 3s ease-in-out;"></div>
        </div>
        <slot name="icon"></slot>
      </div>
    </div>
    <span aria-live="polite" class="progress-bar__aria-live"></span>
  </mock:shadow-root>
  Label
</scale-progress-bar>
`;

exports[`ProgressBar should match snapshot with percentage set 1`] = `
<scale-progress-bar percentage="18">
  <mock:shadow-root>
    <style>
      @keyframes showProgress {
      from {
        width: 0;
      }
      to {
        width: 18%;
      }
    }
    </style>
    <div class="progress-bar" part="progress-bar">
      <div class="progress-bar-wrapper" part="wrapper">
        <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="18" aria-valuetext="18%" class="progress-bar__outer" id="progress-bar-4" part="outer" role="progressbar" style="height: 6px;">
          <div class="progress-bar__inner" part="inner" style="width: 18%; border: 1px solid transparent; animation: showProgress 3s ease-in-out;"></div>
        </div>
        <slot name="icon"></slot>
      </div>
    </div>
    <span aria-live="polite" class="progress-bar__aria-live">
      20%
    </span>
  </mock:shadow-root>
  Label
</scale-progress-bar>
`;

exports[`ProgressBar should match snapshot with statusDescription set 1`] = `
<scale-progress-bar status-description="statusDescription">
  <mock:shadow-root>
    <style>
      @keyframes showProgress {
      from {
        width: 0;
      }
      to {
        width: 0%;
      }
    }
    </style>
    <div class="progress-bar" part="progress-bar">
      <div class="progress-bar-wrapper" part="wrapper">
        <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="0" aria-valuetext="0%" class="progress-bar__outer" id="progress-bar-2" part="outer" role="progressbar" style="height: 6px;">
          <div class="progress-bar__inner" part="inner" style="width: 0%; border: 1px solid transparent; animation: showProgress 3s ease-in-out;"></div>
        </div>
        <slot name="icon"></slot>
      </div>
    </div>
    <div class="progress-bar__status-description" part="status-description" role="alert">
      statusDescription
    </div>
    <span aria-live="polite" class="progress-bar__aria-live"></span>
  </mock:shadow-root>
  Label
</scale-progress-bar>
`;

exports[`ProgressBar should match snapshot with statusInside set 1`] = `
<scale-progress-bar status-inside="">
  <mock:shadow-root>
    <style>
      @keyframes showProgress {
      from {
        width: 0;
      }
      to {
        width: 0%;
      }
    }
    </style>
    <div class="progress-bar" part="progress-bar">
      <div class="progress-bar-wrapper" part="wrapper">
        <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="0" aria-valuetext="0%" class="progress-bar__outer" id="progress-bar-3" part="outer" role="progressbar" style="height: 6px;">
          <div class="progress-bar__inner" part="inner" style="width: 0%; border: 1px solid transparent; animation: showProgress 3s ease-in-out;">
            <div aria-hidden="true" class="progress-bar__inner-status" part="inner-status">
              0%
            </div>
          </div>
        </div>
        <slot name="icon"></slot>
      </div>
    </div>
    <span aria-live="polite" class="progress-bar__aria-live"></span>
  </mock:shadow-root>
  Label
</scale-progress-bar>
`;
